src/HtmlGenerator/Program.cs

```diff
@@ -311,6 +332,17 @@ private static IEnumerable<string> GetAssemblyNames(string filePath)
 
             foreach (var path in solutionFilePaths)
             {
+                var solutionFolder = mergedSolutionExplorerRoot;
+
+                if (rootPath is object)
+                {
+                    var relativePath = Paths.MakeRelativeToFolder(Path.GetDirectoryName(path), rootPath);
+                    var segments = relativePath.Split(new[] { Path.DirectorySeparatorChar, Path.AltDirectorySeparatorChar }, StringSplitOptions.RemoveEmptyEntries);
+
+                    foreach (var segment in segments)
+                        solutionFolder = solutionFolder.GetOrCreateFolder(segment);

```

Comment:
Could you add a block statement? Ideally I should specify this in the .editorconfig.